src/crimson/common/operation.h

```diff
@@ -502,15 +502,22 @@ class PipelineHandle {
   }
 
   template <typename OpT, typename T>
-  seastar::future<>
-  do_enter(T &stage, typename T::BlockingEvent::template Trigger<OpT>&& t) {
-    auto fut = t.maybe_record_blocking(stage.enter(t), stage);
-    exit();
-    return std::move(fut).then(
-      [this, t=std::move(t)](auto &&barrier_ref) mutable {
+  std::optional<seastar::future<>>
+  do_enter_maybe_sync(T &stage, typename T::BlockingEvent::template Trigger<OpT>&& t) {
+    if constexpr (T::is_enter_async) {
```

Comment:
nit: is_blocking_enter

Reply:
changed to `is_enter_sync` following the suggestion around introducing a safer version of `enter_stage_sync()`